halium-chroot

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+#
+# This script will:
+# - mount rootfs.img, system.img and /data
+# - enter an interactive shell in the chroot or execute
+#   a given command
+# - umount rootfs, system and /data
+#
+# Usage:
+#
+# To get an interactive shell in the chroot simply enter:
+# ./halium-chroot
+#
+# To directly execute a non-interactive program run:
+# ./halium-chroot my-program
+#
+
+mount_all()
+{
+	adb shell <<-SHELL
+		mkdir -p /target
+		mount /data/rootfs.img /target
+		mount /data/system.img /target/system
+		mount --bind /data /target/data
+		exit
+	SHELL
+}
+
+umount_all()
+{
+	adb shell <<-SHELL
+		umount -rl /target
+		exit
+	SHELL
+}
+
+mount_all > /dev/null
+
+CHROOT='TERM=linux HOME=/root PATH=/sbin:/bin:/usr/sbin:/usr/bin:$PATH chroot /target'
+LINES=$(tput lines)
+COLUMNS=$(tput cols)
+
+if [ $# -eq 0 ]
+then # Interactive shell
+	if [ $(which tmux) ]
+	then # with tmux installed
+		tmux new-session -s halium-adb -d 'adb shell'
```

Comment:
This won't probably give me shell if I am already in the tmux session :-P

Reply:
Well that is true :-) but without a proper terminal multiplexer it is a pain to get an interactive shell where you can inject the commands to start the chroot. Maybe I can detect an already running session and work around that.

Reply:
yeah, would be lovely.. :)